torch/_inductor/choices.py

```diff
@@ -52,6 +53,34 @@ class Sortable(typing.Protocol):
     def __lt__(self, other: typing.Self) -> bool: ...
 
 
+@dataclasses.dataclass
+class FusionScore:
+    template_score: int
+    node_type_score: bool
+    memory_score: int
+    proximity_score: int
+
+    def __lt__(self, other):
+        """
+        node_type_score has higher priority than memory_score unless
+        the memory_score differs too much
+        """
+        threshold = 16
+        if self.template_score != other.template_score:
+            return self.template_score < other.template_score
+
+        if self.memory_score > other.memory_score * threshold:
```

Comment:
```suggestion
        if min(self.memory_score, other.memory_score * threshold) < max(self.memory_score, other.memory_score):
            return self.memory_score < other.memory_score
```

nit; seems more clear